From b826bd497fc431dcced3f1b0070e8a931145800e Mon Sep 17 00:00:00 2001
From: lei shen <lei.shen@intel.com>
Date: Mon, 16 Nov 2015 11:37:35 +0800
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: When scaling_src_size changed, pf
 registers need to be programmed.

When scaling src size changed, pf registers should be programmed, or
display will hang and show vblank time out issue.

	SQUASHME! - This patch should be squashed into the following patch
		Author: Arun R Murthy <arun.r.murthy@intel.com>
		Change-Id: I92651ed84d9d3f85fdcabbadd9cbb2f565483e11
		SQUASHME! [VPG]: drm/i915: Enable panel fitter only on HWC request

Change-Id: I026826d60900d47f8a1c6e1dbeaf4aa0c0b54255
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9734
Signed-off-by: lei shen <lei.shen@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 6147378..bcbdaad 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11206,6 +11206,7 @@ int intel_set_disp_calc_flip(struct drm_mode_set_display *disp,
 	if (disp->update_flag & DRM_MODE_SET_DISPLAY_UPDATE_PANEL_FITTER) {
 		u32 pfitcontrol = I915_READ(PFIT_CONTROL);
 		u32 pfit_control = pfitcontrol;
+		u32 scaling_src_size = I915_READ(PIPESRC(intel_crtc->pipe));
 
 		/* Check if panel fitter is already enabled on another pipe */
 		if (((pfit_control & PFIT_PIPE_MASK) >> PFIT_PIPE_SHIFT)
@@ -11272,7 +11273,7 @@ int intel_set_disp_calc_flip(struct drm_mode_set_display *disp,
 				pfit_control |= pfit_mode;
 			}
 			intel_crtc->pfit_control = pfit_control;
-			if (pfit_control != pfitcontrol)
+			if (pfit_control != pfitcontrol || scaling_src_size != intel_crtc->scaling_src_size)
 				dev_priv->pfit_changed = true;
 		}
 
-- 
1.9.1

